src/profile_manager/tests/test_views.py

```diff
@@ -176,6 +176,70 @@ def test_profile_detail__student_marks_button(self):
         response = self.client.get(reverse('profiles:profile_detail', args=[s_pk]))
         self.assertContains(response, 'View your Mark Calculations')
 
+    def test_profile_detail__xp_earned_by_tag_show_all_tags_on_profiles_true(self):
+        """ check if 'SiteConfig.get().show_all_tags_on_profiles = True' has ProfileDetail shows all the tags """
+        quest_count = 3
+
+        # for sanity check (its true by default)
+        site_config = SiteConfig.get()
+        site_config.show_all_tags_on_profiles = True
+        site_config.save()
+
+        quest_set = baker.make('quest_manager.quest', _quantity=quest_count)
+
+        # add unique tag for each quest
+        [quest.tags.add(f'TAG-{str(count)}') for count, quest in enumerate(quest_set)]
+
+        # have a submission for one of the quests
+        baker.make(
+            'quest_manager.questsubmission',
+            quest=quest_set[0],
+            user=self.test_student1,
+            is_completed=True,
+            is_approved=True,
+            semester=SiteConfig().get().active_semester,
+        )
+
+        # should have all tags in (TAG-0, TAG-1, TAG-2, ..., TAG-quest_count)
+        self.client.force_login(self.test_student1)
+        response = self.client.get(reverse('profiles:profile_detail', args=[self.test_student1.profile.pk]))
+
+        # should contains all tags including the ones that don't have a submission from the user
+        for i in range(quest_count):
+            self.assertContains(response, f'TAG-{str(i)}')
+
+    def test_xp_earned_by_tag__show_all_tags_on_profiles_false(self):
```

Comment:
Rather than duplicate all this code, just include it in the previous test (and update test name and docstring to account).

Then you can just change the setting at the bottom of the test and recheck the results, instead of re-writing all that duplicate code.  It's also ore efficient because each addition test has overhead.  (ideally everything is it's own test, like this, but our test suite already take like 2 5minutes to run...)

Reply:
done